File: assign4/f#/src/GuessTheNumber.fs
module game 

open System
open src.GuessGame

let seed = System.DateTime.Now.ToString("fff") |> int
let target = getRandomNumber(seed)

let inline readGuess _ = Console.WriteLine("Guess a number between 1 and 100"); Console.ReadLine() |> int
let inline display (message: GuessResult) = Console.WriteLine("Your guess was: {0}", message.ToString())

let attempts = playGame(target, readGuess, display)

Console.WriteLine("Congrats you took {0} attempts to guess correctly", attempts)

File: assign4/javascript/src/guessTheNumber.js
'use strict';

const { getRandomNumber, playGame } = require('./guess_game.js');
const prompt = require('prompt-sync')({ sigint: true });

const target = getRandomNumber(Date.now());
const readGuess = () => prompt('Guess a number between 1 and 100: ');
const display = (result) => { console.log('Your guess was: ' + result.toLowerCase()); };

const attempts = playGame(target, readGuess, display);

console.log('Congrats you took ' + attempts + ' attempts to guess correctly');
